modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/sidebar/SidebarBody.tsx

```diff
@@ -0,0 +1,305 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayPanel from '@clayui/panel';
+import React, {
+	Dispatch,
+	SetStateAction,
+	useCallback,
+	useContext,
+	useEffect,
+	useState,
+} from 'react';
+
+import {EditSchemaContext} from '../EditAPIApplicationContext';
+import BaseAPISchemaProperty from '../baseComponents/BaseAPISchemaProperty';
+import {fetchJSON} from '../utils/fetchUtil';
+
+interface AddedObjectField extends ObjectField {
+	added?: boolean;
+}
+
+interface ObjectDefinitionWithAddedField extends AddedObjectDefinition {
+	objectFields: AddedObjectField[];
+}
+
+interface ObjectFieldsPanelProps {
+	defaultExpanded: boolean;
+	navigate: (id: number) => void;
+	objectDefinition: AddedObjectDefinition;
+	objectRelationshipName?: string;
+	parentDefinitionId: number;
+	schemaUIData: APISchemaUIData;
+	searchKeyword: string;
+	setNavHistory: Dispatch<SetStateAction<AddedObjectDefinition[][]>>;
+	setSchemaUIData: Dispatch<SetStateAction<APISchemaUIData>>;
+	startExpanded?: boolean;
+}
+
+interface SidebarBodyProps {
+	fectchedObjectDefinitions: ObjectDefinitionsRelationshipTree;
+	navHistory: AddedObjectDefinition[][];
+	schemaUIData: APISchemaUIData;
+	searchKeyword: string;
+	setNavHistory: Dispatch<SetStateAction<AddedObjectDefinition[][]>>;
+	setOnBackClick: Dispatch<SetStateAction<voidReturn>>;
+	setSchemaUIData: Dispatch<SetStateAction<APISchemaUIData>>;
+	viewRelatedObjects: boolean;
+}
+
+function ObjectFieldsPanel({
+	defaultExpanded,
+	navigate,
+	objectDefinition,
+	objectRelationshipName,
+	schemaUIData,
+	searchKeyword,
+	setSchemaUIData,
+}: ObjectFieldsPanelProps) {
+	const {
+		fetchedSchemaData,
+		objectDefinitionBasePath,
+		setFetchedSchemaData,
+	} = useContext(EditSchemaContext);
+
+	const [showOnClick, setShowOnClick] = useState<undefined | {id: number}>();
+	const [expanded, setExpanded] = useState(defaultExpanded);
+	const [localUIData, setLocalUIData] = useState<
+		ObjectDefinitionWithAddedField
+	>(objectDefinition);
+
+	const getFilteredFields = (): AddedObjectField[] => {
+		return localUIData.objectFields.filter((field) =>
+			field.label[Liferay.ThemeDisplay.getDefaultLanguageId()]
+				?.toLocaleLowerCase()
+				.includes(searchKeyword.toLocaleLowerCase())
+		);
+	};
+
+	async function getRelationshipsDefinitions(
```

Comment:
@cgmonte `getObjectRelationshipsObjectDefinitions` or `getRelatedObjectDefinitions`. What do you think?

Reply:
getRelatedObjectDefinitions, for brevity :)